upcloud/resource_upcloud_router_test.go

```diff
@@ -90,6 +90,109 @@ func TestAccUpCloudRouter_import(t *testing.T) {
 	})
 }
 
+func TestAccUpCloudRouter_detach(t *testing.T) {
+	var providers []*schema.Provider
+	var router upcloud.Router
+	var network upcloud.Network
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories(&providers),
+		CheckDestroy:      testAccCheckRouterNetworkDestroy,
+		Steps: []resource.TestStep{
+			{
+				// first create network and router attached
+				Config: testAccRouterNetworkConfig("testrouter", "testnetwork", true, true),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRouterExists("upcloud_router.terraform_test_router", &router),
+					testAccCheckNetworkExists("upcloud_network.terraform_test_network", &network),
```

Comment:
can we check if the different computed attributes are set? with `resource.TestCheckResourceAttrSet()`.
Same every where were there are attributes set

Reply:
This turned out to be difficult, due to a circular dependency/chicken-and-egg issue; 
- due to network being dependent on the router (due to the reference), router gets created first. at this point, the computed attribute (attached_networks) is empty.
- network gets created with attachment, but the router resource has no way of knowing of the change, so internally it still has an empty attached_networks
- this will not change until we reimport/update the router and there's no way to trigger such a reimport from another resource (due to the design of tf, which is probably due to wanting to avoid the possibility of circular dependencies)